snapcraft/extensions/kde_neon_6.py

```diff
@@ -109,6 +118,30 @@ def kde_snaps(self) -> KDESnaps6:
         qt6_sdk_snap = _QT6_SDK_SNAP[base]
         kf6_sdk_snap = _KF6_SDK_SNAP[base]
 
+        match base:
+            case "core22":
+                gpu_plugs = {}
+                gpu_layouts = {
+                    "/usr/share/libdrm": {"bind": "$SNAP/kf6-core22/usr/share/libdrm"},
+                }
+            case "core24":
+                gpu_plugs = {
+                    "gpu-2404": {
+                        "interface": "content",
+                        "target": "$SNAP/gpu-2404",
+                        "default-provider": "mesa-2404",
+                    },
+                }
+                gpu_layouts = {
```

Comment:
This isn't really about this PR, but comparing this to the gnome extension we've got a lot of overlap. I wonder whether it would be worth creating an abstract DesktopExtension where this sort of general stuff could live, so we can share more common work?

@mr-cal @sergiusens @kenvandine what do you think?

Things that overlap and could just be extended for each desktop include

- plugs
- GPU plugs
- GPU layouts
- Some base themes

I'm happy to do the work for that, but I only want to do it if I'm actually making things better :-)